GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -154,7 +159,53 @@ protected virtual void ClearNodes()
 
             protected virtual void FillTreeViewNode()
             {
+                var expandedNodes = new HashSet<string>();
+                CollectExpandedNodes(TreeViewNode, expandedNodes);
+
                 Nodes.FillTreeViewNode(TreeViewNode);
+
+                ApplyExpandedNodes(TreeViewNode, expandedNodes);
```

Comment:
It took me few seconds to understand that `CollectExpandedNodes` remembers all expanded nodes to apply expanded state once the tree is rebuilt. For me `collect` associates with garbage collector and objects disposal...
Could we consider renaming these methods to something like: `RememberExpadedState(...)` and `RestoreExpandedState(...)`?

They do look like good candidate for extension method conversion, so we could apply these to any treeview control.
I'd like to see tests for these too please.

Reply:
I would put them in gitGui project. I would create a folder called extensions and a file called TreeviewExtensions. Then if we extend other controls create files so it is organized.

Reply:
> Yes, or perhaps Save/RestoreExpandedState?

👍 

> Not sure what the usual practice is for extension methods. Where should I put them?

Like Jay said, they logically belong to GitUI project.
Right now we don't have a single place (like an `Extensions` folder), and extensions typically live in a separate file next to the extended definition.
In this instance I'd put them into a `TreeViewExtensions.cs` file and have the file in the same folder where the `NativeTreeView.cs` is.
I'd like to consolidate extensions in one place, but it would be a separate story from this one.

Reply:
I think I'm going to have the first function just return the state, rather than have you pass it in, so in that case, instead of "SaveExpandedState", I think just "GetExpandedNodesState" would be better. So the call site will look like:

```csharp
                var expandedNodesState = TreeViewNode.GetExpandedNodesState();
                Nodes.FillTreeViewNode(TreeViewNode);
                TreeViewNode.RestoreExpandedNodesState(expandedNodesState);
```

What I also like about this is that the return type isn't important. We can just use var, and if we want, we can easily change its type in the future.